Bug: Panic in using metrics for lifecycle events

## Steps to reproduce the bug

1. Enable the plugin and link any channel
2. Wait for a lifecycle change notification from MS Teams
3. The plugin will panic with the following error:
```
panic: inconsistent label cardinality: expected 1 label values but got 2 in prometheus.Labels{"discarded_reason":"failed_subscription_check", "event_type":"reauthorizationRequired"} caller="plugin/hclog_adapter.go:54" plugin_id=com.mattermost.msteams-sync
debug [2023-11-23 16:47:12.357 +05:30]                                               caller="plugin/hclog_adapter.go:54" plugin_id=com.mattermost.msteams-sync
debug [2023-11-23 16:47:12.357 +05:30] goroutine 153 [running]:                      caller="plugin/hclog_adapter.go:54" plugin_id=com.mattermost.msteams-sync
debug [2023-11-23 16:47:12.357 +05:30] github.com/prometheus/client_golang/prometheus.(*CounterVec).With(...) caller="plugin/hclog_adapter.go:54" plugin_id=com.mattermost.msteams-sync
debug [2023-11-23 16:47:12.357 +05:30] 	github.com/prometheus/client_golang@v1.14.0/prometheus/counter.go:266 caller="plugin/hclog_adapter.go:54" plugin_id=com.mattermost.msteams-sync
debug [2023-11-23 16:47:12.357 +05:30] github.com/mattermost/mattermost-plugin-msteams-sync/server/metrics.(*metrics).ObserveLifecycleEvent(0xc0001b5b30, {0xc000628708, 0x17}, {0x4995e48, 0x19}) caller="plugin/hclog_adapter.go:54" plugin_id=com.mattermost.msteams-sync
debug [2023-11-23 16:47:12.357 +05:30] 	github.com/mattermost/mattermost-plugin-msteams-sync/server/metrics/metrics.go:353 +0x115 caller="plugin/hclog_adapter.go:54" plugin_id=com.mattermost.msteams-sync
debug [2023-11-23 16:47:12.357 +05:30] github.com/mattermost/mattermost-plugin-msteams-sync/server/handlers.(*ActivityHandler).HandleLifecycleEvent(0xc0004bc410, {{0xc00015c680, 0x32}, {0xc0009dcc40, 0x20}, {0x0, 0x0}, {0xc000628708, 0x17}, {0x0, ...}, ...}) caller="plugin/hclog_adapter.go:54" plugin_id=com.mattermost.msteams-sync
debug [2023-11-23 16:47:12.357 +05:30] 	github.com/mattermost/mattermost-plugin-msteams-sync/server/handlers/handlers.go:139 +0x43d caller="plugin/hclog_adapter.go:54" plugin_id=com.mattermost.msteams-sync
```

## Expected behaviour
There should be no panics.

## Actual behaviour
Plugin panicking while receiving lifecycle change notifications.

## Mattermost Version and other Platforms

 - Mattermost Version: Latest master branch
 - Browser and OS: Chrome on Linux

--- a/server/metrics/metrics.go
+++ b/server/metrics/metrics.go
@@ -190,7 +190,7 @@ func NewMetrics(info InstanceInfo) Metrics {
 		Name:        "lifecycle_events_total",
 		Help:        "The total number of MS Teams lifecycle events received.",
 		ConstLabels: additionalLabels,
-	}, []string{"event_type"})
+	}, []string{"event_type", "discarded_reason"})
 	m.registry.MustRegister(m.lifecycleEventsTotal)
 
 	m.messagesTotal = prometheus.NewCounterVec(prometheus.CounterOpts{

